doc/api/net.md

```diff
@@ -246,168 +352,217 @@ server.on('error', (e) => {
+});
+```
 
 ### Event: 'data'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Buffer}
+The `'data'` event is emitted when data is received.
 
-Emitted when data is received.  The argument `data` will be a `Buffer` or
-`String`.  Encoding of data is set by `socket.setEncoding()`.
-(See the [Readable Stream][] section for more information.)
+The listener callback will be invoked with a single `data` argument that will
+be either a `Buffer` or string (if an encoding is set using
+`socket.setEncoding()`). See the [Readable][] stream documentation for more
+information.
 
-Note that the __data will be lost__ if there is no listener when a `Socket`
+*Note*: __Data will be lost__ if there is no listener when a `net.Socket`
 emits a `'data'` event.
 
 ### Event: 'drain'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when the write buffer becomes empty. Can be used to throttle uploads.
-
-See also: the return values of `socket.write()`
+The `'drain'` event is emitted when the `net.Socket` instances write buffer
+becomes empty. This can be used to throttle writes to the socket. See the
+[Writable][] stream documentation for more information.
 
 ### Event: 'end'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when the other end of the socket sends a FIN packet.
+The `'end'` event is emitted when the other end of the socket sends a `FIN`
+packet.
 
-By default (`allowHalfOpen == false`) the socket will destroy its file
-descriptor  once it has written out its pending write queue.  However, by
-setting `allowHalfOpen == true` the socket will not automatically `end()`
-its side allowing the user to write arbitrary amounts of data, with the
-caveat that the user is required to `end()` their side now.
+By default (when the `allowHalfOpen` option is `false`) the socket will destroy
+its underlying file descriptor once it has completely written out its pending
+write queue.  However, by setting `allowHalfOpen` to `true`, the socket will not
+automatically end its side of the socket allowing additional data to be written.
+
+The listener callback is called without passing any arguments.
+
+```js
+socket.on('end', () => {
+  console.log('The socket has ended!');
+});
+```
 
 ### Event: 'error'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Error}
+The `'error'` event is emitted when an error occurs. The `'close'` event will be
+emitted immediately following this event.
 
-Emitted when an error occurs.  The `'close'` event will be called directly
-following this event.
+The listener callback is called with the `Error` passed as the only argument.
+
+```js
+socket.on('error', (error) => {
+  console.log('There was an error!', error);
+});
+```
 
 ### Event: 'lookup'
 <!-- YAML
 added: v0.11.3
 -->
 
-Emitted after resolving the hostname but before connecting.
-Not applicable to UNIX sockets.
+The `'lookup'` event is emitted after the `net.Socket` has resolved the
+`hostname` given when connection, but *before* the connection is established.
+This event is not applicable when using UNIX sockets.
```

Comment:
is not *emitted* when using local sockets, because no address resolution is required.